src/I18n/Time.php

```diff
@@ -45,6 +45,22 @@ class Time extends Carbon implements JsonSerializable
     protected static $_toStringFormat = [IntlDateFormatter::SHORT, IntlDateFormatter::SHORT];
 
     /**
+     * The format to use when when converting this object to json
+     *
+     * The format should be either the formatting constants from IntlDateFormatter as
+     * described in (http://www.php.net/manual/en/class.intldateformatter.php) or a pattern
+     * as specified in (http://www.icu-project.org/apiref/icu4c/classSimpleDateFormat.html#details)
+     *
+     * It is possible to provide an array of 2 constants. In this case, the first position
+     * will be used for formatting the date part of the object and the second position
+     * will be used to format the time part.
+     *
+     * @var mixed
+     * @see \Cake\I18n\Time::i18nFormat()
+     */
+    protected static $_jsonEncodeFormat = "yyyy-MM-dd'T'HH:mm:ssxx";
```

Comment:
Sorry but can you please be more specific? I changed it to be compatible
with setting toString format. The constant used before will not work with
that. And if I missee the timezone part, I guess tests would not pass.

On Friday, August 14, 2015, José Lorenzo Rodríguez <notifications@github.com>
wrote:

> In src/I18n/Time.php
> <https://github.com/cakephp/cakephp/pull/7231#discussion_r37076028>:
>
> > @@ -45,6 +45,22 @@ class Time extends Carbon implements JsonSerializable
> >      protected static $_toStringFormat = [IntlDateFormatter::SHORT, IntlDateFormatter::SHORT];
> >
> >      /**
> > +     * The format to use when when converting this object to json
> > +     *
> > +     * The format should be either the formatting constants from IntlDateFormatter as
> > +     * described in (http://www.php.net/manual/en/class.intldateformatter.php) or a pattern
> > +     * as specified in (http://www.icu-project.org/apiref/icu4c/classSimpleDateFormat.html#details)
> > +     *
> > +     * It is possible to provide an array of 2 constants. In this case, the first position
> > +     * will be used for formatting the date part of the object and the second position
> > +     * will be used to format the time part.
> > +     *
> > +     * @var mixed
> > +     * @see \Cake\I18n\Time::i18nFormat()
> > +     */
> > +    protected static $_jsonEncodeFormat = "yyyy-MM-dd'T'HH:mm:ssxx";
>
> I think you can reference the DateTime constant that was used before in
> the method. Otherwise you miss the timezone part
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cakephp/cakephp/pull/7231/files#r37076028>.
>